File: src/Client.UWP/View/LandUnitStrategy.cs
namespace Client.View
{
    public class LandUnitStrategy : ITileStrategy
    {
        private readonly TextureHolder texture;

        public LandUnitStrategy(TextureHolder landUnitTexture)
        {
            texture = landUnitTexture;
        }

        public bool CanExecute(LocationType[] neighbors)
        {
            return neighbors[Directions.NeighborThis] == LocationType.LandUnit;
        }

        public TextureHolder Execute(LocationType[] neighbors)
        {
            return texture;
        }
    }
}

File: core/src/com/mygdx/game/view/LandUnitStrategy.java
package com.mygdx.game.view;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.mygdx.game.resources.LandUnitTextures;

@Singleton
public final class LandUnitStrategy implements ITileStrategy {

    private final LandUnitTextures textures;

    @Inject
    public LandUnitStrategy(LandUnitTextures landUnitTexture) {
        textures = landUnitTexture;
    }

    public boolean canExecute(LocationType[] neighbors) {
        return neighbors[Directions.NeighborThis] == LocationType.LandUnit;
    }

    public TextureHolder execute(LocationType[] neighbors) {
        return textures.getTexture();
    }
}
